tests/Query/Filter/PriorityField.test.ts

```diff
@@ -67,6 +67,24 @@ describe('priority below', () => {
     });
 });
 
+describe('priority is not', () => {
+    it('priority is not none', () => {
+        const filter = 'priority is not none';
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, false);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.High, true);
+    });
+
+    it('priority is not high', () => {
+        const filter = 'priority is not high';
+        testTaskFilterForTaskWithPriority(filter, Priority.Low, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.None, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.Medium, true);
+        testTaskFilterForTaskWithPriority(filter, Priority.High, false);
+    });
+});
+
```

Comment:
I don't like how repetitive this is. I do feel that it's important that tests are very clear in their intention.
But is there a way to make this just as clear, but less repetitive.
I wondered about some kind of dict-type structure that just lists the 4 priority values, and the 4 true/false expectations.
Am unsure about how to write it, or if it's worthwhile.

Reply:
What about this?
```ts
    it('priority is not none', () => {
        Object.values(Priority).forEach(
            priority => testTaskFilterForTaskWithPriority(
                'priority is not none', priority, priority !== Priority.None)
        );
    });
```

Reply:
Thanks. That is a nice idea. I really haven't internalised all that can be done with `.forEach()`

My slight concern about that is that I think it is a copy of the code being tested. If that is true, then I worry whether it's a very robust test.

Reply:
You could also use parametrization to test all combinations:
```ts
    it.each([['low', Priority.Low], ['none', Priority.None],
             ['medium', Priority.Medium], ['high', Priority.High]])
    ('priority is not %s', (keyword: string, priority: Priority) => {
        for (const p of Object.values(Priority))
            testTaskFilterForTaskWithPriority(
                    `priority is not ${keyword}`, p, p !== priority);
    });
```
But yes, the `p !== priority` bit again mimics part of the tested code. If you want to avoid that, you can specify a large enough number of sample input and expected values instead:
```ts
    it.each([['low', Priority.Low, false],
             ['low', Priority.None, true],
             ['none', Priority.None, false],
             ['none', Priority.Medium, true],
             ['medium', Priority.None, true],
             ['medium', Priority.Medium, false],
             ['high', Priority.Medium, true],
             ['high', Priority.High, false]])
    ('priority is not %s (with %s)', (filter: string, input: Priority, expected: boolean) => {
        testTaskFilterForTaskWithPriority(
                `priority is not ${filter}`, input, expected);
    });
```

Reply:
Thank you. I went for your last suggestion:

https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1365/commits/36a7ec4d647d59995678002f92972886a2bb1dc9

I like the code, but the use of numbers instead of text for the enum values makes it a little harder to understand what's going on the tests.

![image](https://user-images.githubusercontent.com/4840096/205745963-0c32b893-966e-414d-a631-2e8d0b5fa851.png)


Having said that, at least each test is named now, whereas in my one, if a test failed, it could have been any one of 4 of the tests that failed.